app/org/sagebionetworks/bridge/models/schedules/ScheduleCriteria.java

```diff
@@ -5,60 +5,68 @@
 
 import org.sagebionetworks.bridge.models.Criteria;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.google.common.collect.Sets;
 
+@JsonDeserialize(builder=ScheduleCriteria.Builder.class)
 public final class ScheduleCriteria implements Criteria {
     private final Schedule schedule;
-    private final Integer minAppVersion;
-    private final Integer maxAppVersion;
-    private final Set<String> allOfGroups;
-    private final Set<String> noneOfGroups;
+    private final Criteria criteria;
```

Comment:
Very true, in this case I can just get rid of all the intermediate stuff and remove the interface.